include/deal.II/physics/elasticity/standard_tensors.h

```diff
@@ -0,0 +1,360 @@
+       * it may be strictly applied through the chain rule to achieve the
+       * definition of the second Piola-Kirchhoff stress, i.e.
+       * @f[
+       *   \mathbf{S}
+       *     = 2\frac{\partial \psi \left( \bar{\mathbf{C}} \right)}{\partial \mathbf{C}}
+       *     = 2\frac{\partial \psi \left( \bar{\mathbf{C}} \right)}{\partial \bar{\mathbf{C}}}
+       *     : \frac{\partial \bar{\mathbf{C}}}{\partial \mathbf{C}}
+       *     = \bar{\mathbf{S}} : \hat{\mathcal{P}}
+       *     \equiv \hat{\mathcal{P}}^{T} : \bar{\mathbf{S}} \, .
+       * @f]
+       *
+       * @note Comparing the definition of this tensor in Holzapfel (2001) to that
+       * adopted here, the inclusion of the extra factor $J^{-2/\textrm{dim}}$ does not,
+       * at the outset, seem to be a reasonable choice.
+       * However, in the author's view it makes direct implementation of the
+       * expressions for isochoric (referential) stress contributions and their
+       * linearisation simpler in practise.
+       */
+      template <typename Number>
+      static SymmetricTensor<4, dim, Number>
+      Dev_P (const Tensor<2, dim, Number> &F);
+
+      /**
+       * Returns the transpose of the fourth-order referential deviatoric tensor,
+       * as constructed from the deformation gradient tensor @p F.
+       * The result performs the following operation:
+       * @f[
+       *  \textrm{Dev}^{T} \{ \bullet \} = J^{-2/\textrm{dim}} \left[ \{ \bullet \} -
+       *  \frac{1}{\textrm{dim}} \left[\mathbf{C}^{-1} : \{ \bullet \}\right] \mathbf{C} \right]
+       *  := \hat{\mathcal{P}}^{T} : \{ \bullet \}
+       * @f]
+       */
+      template <typename Number>
+      static SymmetricTensor<4, dim, Number>
+      Dev_P_T (const Tensor<2, dim, Number> &F);
+
+//@}
+
+      /**
+       * @name Scalar derivatives
+       */
+//@{
+      /**
+       * Returns the derivative of the volumetric Jacobian
+       * $J = \text{det} \mathbf{F}$ with respect to the right Cauchy-Green
+       * tensor, as constructed from the deformation gradient tensor @p F.
+       * The computed result is
+       * @f[
+       *  \frac{\partial J}{\partial \mathbf{C}}
+       *   = \frac{1}{2} J \mathbf{C}^{-1}
+       * @f]
+       */
+      template <typename Number>
+      static SymmetricTensor<2, dim, Number>
+      ddet_F_dC (const Tensor<2, dim, Number> &F);
+
+//@}
+
+      /**
+       * @name Tensor derivatives
+       */
+//@{
+
+      /**
+       * Returns the derivative of the inverse of the right Cauchy-Green
+       * tensor with respect to the right Cauchy-Green tensor itself,
+       * as constructed from the deformation gradient tensor @p F.
+       * The result, accounting for symmetry, is defined in index notation as
+       * @f[
+       *  \left[ \frac{\partial \mathbf{C}^{-1}}{\partial \mathbf{C}} \right]_{abcd}
+       *    := -\frac{1}{2}[ C^{-1}_{ac}C^{-1}_{bd}
+       *     + C^{-1}_{ad}C^{-1}_{bc}  ]
```

Comment:
This is just the usual index notation. There is an abnormal combination of indices on the RHS, so thats why I used it. To the chosen indices {a,b,c,d} bother you?